### PR TITLE
Enable more rules in codebase

### PR DESCRIPTION
**Pull request by adidahiya**
See commit messages

### CODE REVIEW

**Comment by mohsen1** [comment]
Also add import-spacing and typeof-compare to base?

**Comment by adidahiya** [comment]
@mohsen1 I only want to enable rules that have been released -- we use the latest TSLint release to lint our source code, not the current binary in `./bin/tslint`

File: src/rules/importBlacklistRule.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ class NoRequireFullLibraryWalker extends Lint.RuleWalker {
     }
 
     public visitImportEqualsDeclaration(node: ts.ImportEqualsDeclaration) {
-        let moduleReference = <ts.ExternalModuleReference> node.moduleReference;
+        const moduleReference = <ts.ExternalModuleReference> node.moduleReference;
```
Comment (nchen63) [inline src/rules/importBlacklistRule.ts]: were these not caught by the auto fixer?

Comment (adidahiya) [inline src/rules/importBlacklistRule.ts]: you mean the angle bracket assertion? this is an outdated diff; I did that in the other PR

**Pull request merged**